### DISCUSSION TITLE
Netbeans 22 with Java 21 using SQL Profiler, the profiler does nothing anymore

### DISCUSSION THREAD

**Discussion by sandeepIUOE** [apache/netbeans#7746] [General]
When I try to use the SQL profiler, which used to work for Java 8 and Netbeans 12.5, but when I tried recompiling the same project to Java 21 and Netbeans 22, and tried to run SQL profiler, it doesn't show any SQL queries, and I have to dig into the Netbeans IDE logs to see the below. 

(sample only, full error log is all the same type of error): 

*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/KeyFactory:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/KeyFactorySpi:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ec/ECKeyFactory:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/KeyPairGeneratorSpi:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/KeyPairGenerator:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ec/ECKeyPairGenerator:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/util/ECUtil:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/AlgorithmParameters:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/AlgorithmParametersSpi:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/util/ECParameters:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/util/ECKeySizeParameterSpec:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class java/security/KeyPairGenerator$Delegate:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLContextImpl$CustomizedSSLProtocols:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/EphemeralKeyManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/EphemeralKeyManager$EphemeralKeyPair:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/SSLSessionContext:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSessionContextImpl:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/util/Cache:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/util/MemoryCache:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/KeyManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/X509KeyManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/X509ExtendedKeyManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/DummyX509KeyManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/X509ExtendedTrustManager:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/AbstractTrustManagerWrapper:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/SocketFactory:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/SSLSocketFactory:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketFactoryImpl:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class javax/net/ssl/SSLSocket:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/BaseSSLSocketImpl:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLTransport:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketImpl:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketImpl$AppInputStream:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketImpl$AppOutputStream:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HandshakeHash:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HandshakeHash$TranscriptHash:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HandshakeHash$CacheOnlyHash:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/ConnectionContext:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/TransportContext:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Record:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/InputRecord:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLRecord:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketInputRecord:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLCipher$SSLReadCipher:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator$SSLAuthenticator:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator$SSLNullAuthenticator:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator$MAC:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator$SSLNullMac:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/Authenticator$MacImpl:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLCipher$NullReadCipherGenerator$NullReadCipher:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/OutputRecord:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLSocketOutputRecord:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLConsumer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HandshakeProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLHandshake:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HelloRequest:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/SSLProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HelloRequest$HelloRequestKickstartProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HelloRequest$HelloRequestConsumer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HelloRequest$HelloRequestProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/ClientHello:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/ClientHello$ClientHelloKickstartProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/ClientHello$ClientHelloConsumer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/ClientHello$ClientHelloProducer:
Unsupported class file version: 65.0
*** Profiler warning (Tue Sep 10 16:24:29 EDT 2024): Error reading class sun/security/ssl/HandshakeConsumer:
Unsupported class file version: 65.0

**Comment by JoachimRohde** [comment]
It's a known issue: https://github.com/apache/netbeans/issues/7323

**Reply by sandeepIUOE** [reply]
Do you think this will be fixed? That thread is depressing, they are talking about just removing it from the IDE all together.

**Reply by neilcsmith-net** [reply]
It might be depressing, but it's also a reality if no-one steps up to work on it.  Keeping broken features in the UI is worse.
